op-scim-bridge/templates/secret.yaml

```diff
@@ -18,12 +20,7 @@ metadata:
     {{- toYaml . | nindent 4 }}
   {{- end }}
 type: Opaque
-stringData:
+data:
   {{ .Values.scim.credentialsSecret.key }}: |-
-  {{ if .Values.scim.credentialsSecret.value_json }}
-  {{- .Values.scim.credentialsSecret.value_json | b64enc | indent 2 }}
-  {{ else }}
-  {{- .Values.scim.credentialsSecret.value_base64 | indent 2 }}
-  {{ end }}
-{{- end }}
```

Comment:
We would like to maintain the ability to ingest both json and raw base64, as they each have a place in differing workflows. However what we can do is rename `value_json` to just be `value` or something similarly default. JSON ingestion is the primary use case. 

This would replicate the [credentials ingestion for the 1Password Connect server](https://github.com/1Password/connect-helm-charts/blob/main/charts/connect/templates/connect-credentials.yaml) which has both `connect.credentials` and `connect.credentials_base64`

Reply:
Unfortunately that's an anti-pattern for Helm which is why you won't find the mechanism built in.

However I think you're confusing what happens. The 'json' string can be passed in, as I said in my PR notes, and the k8s API will automatically encode it as a bse64 string for storage in a secret (part of the reason to use secrets, instead of volumes). Taking it in as a file does the same thing, expect this time reading the string from file contents instead of `stdin`.

If you really want to have the ability for people to encode the string themselves (not sure why you would want to do this when it happens anyway, and people are just prone to insert errors in their encoding), then I suggest you make that a mechanism outside this loop, as it breaks a number of core Helm functionality assumptions.

Reply:
This is true.  JSON, PEM-formatted certificates, whatever, k8s will store it as a base64 single line string if you create a secret.  The only thing adding a manual "pre-base64'd" option does is create possible breakage.  Even if you're concerned about writing a migration tool for existing users, reading a file out of a k8s `volume` and writing it to a `secret` can be handled automatically, and if the file is already base64'd, just undo that before calling `create secret`.